tests/nnstreamer_filter_mvncsdk2/NCSDKTensorFilterTestHelper.cc

```diff
@@ -0,0 +1,600 @@
+/**
+ * @file NCSDKTensorFilterTestHelper.cc
+ * @date 8 Jan 2020
+ * @author  Wook Song <wook16.song@samsung.com>
+ * @brief Helper class for testing tensor_filter_mvncsdk2 without an actual device
+ * @see https://github.com/nnsuite/nnstreamer
+ * @bug	No known bugs except for NYI items
+ *
+ *  Copyright 2020 Samsung Electronics
+ *
+ */
+
+#include "NCSDKTensorFilterTestHelper.hh"
+
+/**
+ * @brief Default constuctor. Note that explicit invocation of init () is always
+ * required after getting the instance.
+ */
+NCSDKTensorFilterTestHelper::NCSDKTensorFilterTestHelper ()
+{
+  // MAJOR should be 2
+  this->mVer[MAJOR] = 2;
+  // Don't care other indexes, MINOR, HOTFIX, and RC
+  this->mVer[MINOR] = 3;
+  this->mVer[HOTFIX] = 4;
+  this->mVer[RC] = 5;
+
+  try {
+    this->mDevHandle = new ncDeviceHandle_t;
+  } catch (const std::bad_alloc & e) {
+    this->mDevHandle = nullptr;
+  }
+
+  try {
+    this->mGraphHandle = new ncGraphHandle_t;
+  } catch (const std::bad_alloc & e) {
+    this->mGraphHandle = nullptr;
+  }
+  this->mModelPath = nullptr;
+}
+
+/**
+ * @brief Destructor.Resources are deallocated.
+ */
+NCSDKTensorFilterTestHelper::~NCSDKTensorFilterTestHelper ()
+{
+  /* Do nothing */
+}
+
+/**
+ * @brief A method to initialize this helper class
+ * @param[in] model : A neural network model to simulate
+ */
+void
+NCSDKTensorFilterTestHelper::init (model_t model)
+{
+  switch (model) {
+    default:
+    case GOOGLE_LENET:
+      this->mModel = model;
+      try {
+        this->mTensorDescInput = new struct ncTensorDescriptor_t ();
+        this->mTensorDescInput->c = GOOGLE_LENET_IN_DIM_C;
+        this->mTensorDescInput->n = GOOGLE_LENET_IN_DIM_N;
+        this->mTensorDescInput->w = GOOGLE_LENET_IN_DIM_W;
+        this->mTensorDescInput->h = GOOGLE_LENET_IN_DIM_H;
+        this->mTensorDescInput->dataType = NC_FIFO_FP32;
+
+        this->mTensorDescOutput = new struct ncTensorDescriptor_t ();
+        this->mTensorDescOutput->c = GOOGLE_LENET_OUT_DIM_C;
+        this->mTensorDescOutput->n = GOOGLE_LENET_OUT_DIM_N;
+        this->mTensorDescOutput->w = GOOGLE_LENET_OUT_DIM_W;
+        this->mTensorDescOutput->h = GOOGLE_LENET_OUT_DIM_H;
+        this->mTensorDescOutput->dataType = NC_FIFO_FP32;
+      } catch  (const std::bad_alloc & e) {
+        this->mTensorDescInput = nullptr;
+        this->mTensorDescOutput = nullptr;
+      }
+      break;
+  }
+}
+
+/**
+ * @brief A method to release allocated resources
+ */
+void
+NCSDKTensorFilterTestHelper::release ()
+{
+  this->ncDeviceDestroy (&(this->mDevHandle));
```

Comment:
How about checking `mDevHandle` and `mGraphHandle` because they can be null in the constructor?

Reply:
ncDeviceDestory and ncGraphDestroy already do.

Reply:
I think if `this->mDevHandle` is null, `&(this->mDevHandle)` may cause segmentation fault before calling `this->ncDeviceDestroy()`